planet/http.py

```diff
@@ -265,11 +268,12 @@ async def alog_response(*args, **kwargs):
 
     @classmethod
     async def _raise_for_status(cls, response):
-        try:
-            response.raise_for_status()
-        except httpx.HTTPStatusError as e:
-            await e.response.aread()
-            cls._convert_and_raise(e)
+        if response.is_error:
```

Comment:
See https://github.com/planetlabs/planet-client-python/pull/685#discussion_r961097899. This is probably the same thing as `response.status_code >= 400` but syntactically a little neater, maybe?